Volume/View Models/MagazinesListViewModel.swift

```diff
@@ -0,0 +1,175 @@
+            // TODO: replace logic when backend implements Publication.getMagazineSemesters
+            Network.shared.publisher(
+                for: GetAllMagazineSemestersQuery(limit: 100)
+            )
+            .map { $0.magazines.map(\.semester) }
+            .sink { [weak self] completion in
+                self?.networkState.handleCompletion(screen: .magazines, completion)
+            } receiveValue: { [weak self] semesters in
+                guard let `self` = self else { return }
+                self.allSemesters = Array(Set(semesters))
+                    .sorted(by: self.compareSemesters)
+                
+                self.allSemesters?.reverse()
+                self.allSemesters?.insert("all", at: 0)
+                self.selectedSemester = "all"
+                self.fetchAllMagazines()
+            }
+            .store(in: &queryBag)
+        }
+        
+        func fetchMagazinesBySemester(_ semester: String) {
+            sectionStates.magazinesBySemester = .loading
+
+            sectionQueries.magazinesBySemester = Network.shared.publisher(
+                for: GetMagazinesBySemesterQuery(semester: semester)
+            )
+            .map { $0.magazines.map(\.fragments.magazineFields) }
+            .sink { [weak self] completion in
+                self?.networkState.handleCompletion(screen: .magazines, completion)
+            } receiveValue: { [weak self] magazineFields in
+                Task {
+                    let magazinesBySemester = await [Magazine](magazineFields)
+                    withAnimation(.linear(duration: 0.1)) {
+                        self?.sectionStates.magazinesBySemester = .results(magazinesBySemester)
+                    }
+                }
+            }
+        }
+        
+        func fetchAllMagazines() {
+            sectionStates.magazinesBySemester = .loading
+            sectionQueries.magazinesBySemester = Network.shared.publisher(
+                for: GetAllMagazinesQuery(limit: Constants.allMagazinesLimit, offset: offset)
+            )
+            .map { $0.magazines.map(\.fragments.magazineFields) }
+            .sink { [weak self] completion in
+                self?.networkState.handleCompletion(screen: .magazines, completion)
+            } receiveValue: { [weak self] magazineFields in
+                Task {
+                    let allMagazines = await [Magazine](magazineFields)
+                    withAnimation(.linear(duration: 0.1)) {
+                        self?.sectionStates.magazinesBySemester = .results(allMagazines)
+                    }
+                }
+            }
+        }
+        
+        // MARK: Helpers
+
+        private func compareSemesters(_ s1: String, _ s2: String) -> Bool {
+            if let s1Year = Int(s1.suffix(2)),
+               let s2Year = Int(s2.suffix(2)),
+               s1Year != s2Year {
+                return s1Year < s2Year
+            }
+
+            let validSemesterStrings = ["sp", "fa"]
```

Comment:
consider making this into an enum too, and change the switch statement below accordingly